pallets/roles/src/lib.rs

```diff
@@ -228,33 +215,114 @@ pub mod pallet {
 		/// - `role`: Role to remove from the validator.
 		///
 		/// This function will return error if
+		/// - Account is not a validator account.
 		/// - Role is not assigned to the validator.
+		/// - All the jobs are not completed.
 		#[pallet::weight({0})]
 		#[pallet::call_index(1)]
 		pub fn clear_role(origin: OriginFor<T>, role: RoleType) -> DispatchResult {
 			let stash_account = ensure_signed(origin)?;
+			// Ensure stash account is a validator.
+			ensure!(
+				pallet_staking::Validators::<T>::contains_key(&stash_account),
+				Error::<T>::NotValidator
+			);
+
 			// check if role is assigned.
 			ensure!(
 				Self::is_validator(stash_account.clone(), role.clone()),
-				Error::<T>::RoleNotAssigned
+				Error::<T>::NoRoleAssigned
 			);
 			// TODO: Call jobs manager to remove the services.
 			// On successful removal of services, remove the role from the mapping.
 			// Issue link for reference : https://github.com/webb-tools/tangle/issues/292
 
-			// Unbound locked funds.
-			let ledger = Self::ledger(&stash_account).ok_or(Error::<T>::InvalidStashController)?;
-			Self::unbond(&ledger)?;
-			Self::deposit_event(Event::<T>::Unbonded {
-				account: ledger.stash,
-				amount: ledger.total_locked,
-			});
-
 			// Remove role from the mapping.
 			AccountRolesMapping::<T>::remove(&stash_account);
+			// Remove stash account related info.
+			Self::kill_stash(&stash_account);
+
 			Self::deposit_event(Event::<T>::RoleRemoved { account: stash_account, role });
 
 			Ok(())
 		}
+
+		/// Declare no desire to either validate or nominate.
+		///
+		/// If you have opted for any of the roles, please submit `clear_role` extrinsic to opt out
+		/// of all the services. Once your role is cleared, your request will be processed.
+		///
+		/// # Parameters
+		///
+		/// - `origin`: Origin of the transaction.
+		///
+		/// This function will return error if
+		/// - Account is not a validator account.
+		/// - Role is assigned to the validator.
+		#[pallet::weight({0})]
+		#[pallet::call_index(2)]
+		pub fn chill(origin: OriginFor<T>) -> DispatchResult {
+			let account = ensure_signed(origin.clone())?;
+			// Ensure no role is assigned to the account before chilling.
+			ensure!(!AccountRolesMapping::<T>::contains_key(&account), Error::<T>::HasRoleAssigned);
+
+			// chill
+			let _ = pallet_staking::Pallet::<T>::chill(origin);
+			Ok(())
```

Comment:
```suggestion
			pallet_staking::Pallet::<T>::chill(origin)
```

return the result of the inner call